cmd/collector/app/options.go

```diff
@@ -172,5 +191,15 @@ func (o options) apply(opts ...Option) options {
 	if ret.numWorkers == 0 {
 		ret.numWorkers = DefaultNumWorkers
 	}
+	if ret.dynQueueSizeMemory == 0 {
+		// this seems to be the best way to get the total memory size of the host without resorting to external libraries
```

Comment:
I think if this option is zero the functionality should be disabled. If people want to manage queue size by memory, they should be in charge of defining the limit, not some hidden heuristic in Jaeger.

Reply:
Agree that people should specify that all the time, but this is in place as a "sensible default" that would be effective only when the warmup value is set. I can remove it and disable the feature with a warning in case only the warmup value is set.

Reply:
I prefer not to provide "sensible default" for this setting (which would also make the CGO changes unnecessary). Let people figure out what they want to do with the memory.